data/trie/patriciaMerkleTrie.go

```diff
@@ -517,6 +516,12 @@ func (tr *patriciaMerkleTrie) Snapshot() error {
 
 	if len(tr.snapshots) > maxSnapshots {
 		dbUniqueId := strconv.Itoa(tr.snapshotId - len(tr.snapshots))
+
+		err = tr.snapshots[0].Close()
+		if err != nil {
+			tr.snapshotInProgress = false
+			return err
+		}
 		tr.snapshots = tr.snapshots[1:]
 
 		removePath := path.Join(tr.snapshotDbCfg.FilePath, dbUniqueId)
```

Comment:
do we have a test to validate that we are removing the right db?
otherwise, instead of computing the id based on size and current snapshot id I would store it (or the dbpath) in the snapshot list and just take it from there

Reply:
Yes, there is a test that verifies that the right db was removed. (TestDeleteOldSnapshots() from `node_test.go`)